### DISCUSSION TITLE
Port tests from sigma rules test_rules.py

### DISCUSSION THREAD

**Discussion by frack113** [SigmaHQ/pySigma#158] [Ideas]
List of the tests:

- [X] test_all_of_them_condition
- [ ] test_all_value_modifier_single_item
- [ ] test_condition_operator_casesensitive
- [ ] test_confirm_correct_mitre_tags
- [ ] test_duplicate_detections
- [X] test_duplicate_references
- [X] test_duplicate_tags
- [X] test_duplicate_titles
- [ ] test_event_id_instead_of_process_creation
- [ ] test_field_name_with_space
- [ ] test_field_user_localization
- [ ] test_file_names
- [ ] test_legal_trademark_violations
- [ ] test_level
- [ ] test_look_for_duplicate_filters
- [ ] test_missing_date
- [ ] test_missing_description
- [X] test_missing_id
- [ ] test_optional_author
- [ ] test_optional_date_modified
- [ ] test_optional_falsepositives_blocked_content
- [ ] test_optional_falsepositives_capital
- [ ] test_optional_falsepositives_listtype
- [ ] test_optional_fields
- [ ] test_optional_license
- [ ] test_optional_related
- [ ] test_optional_status
- [X] test_optional_tags
- [ ] test_optional_target
- [ ] test_optional_tlp
- [ ] test_re_invalid_escapes
- [ ] test_references
- [ ] test_references_in_description
- [ ] test_references_plural
- [ ] test_selection_list_one_value
- [ ] test_selection_start_or_and
- [ ] test_single_named_condition_with_x_of_them
- [ ] test_source_eventlog
- [ ] test_sysmon_rule_without_eventid
- [ ] test_title
- [ ] test_title_in_first_line
- [ ] test_unused_selection

**Comment by thomaspatzke** [comment]
I've made [a project](https://github.com/orgs/SigmaHQ/projects/4) from it 😉
